uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/LoginIT.java

```diff
@@ -269,6 +269,13 @@ public void testPasscodeRedirect() {
         attemptLogin(testAccounts.getUserName(), testAccounts.getPassword());
 
         assertThat(webDriver.findElement(By.cssSelector("h1")).getText(), Matchers.containsString("Temporary Authentication Code"));
+
+        // Verify that the CopyToClipboard function can be executed
+        String passcode = webDriver.findElement(By.id("passcode")).getText();
+        ((JavascriptExecutor)webDriver).executeScript("CopyToClipboard",
+                passcode);
+        // Verify that the copybutton can be clicked
+        webDriver.findElement(By.id("copybutton")).click();
```

Comment:
Not knowing what the semantics of the `click()` method is, from its name it looks to me that it just clicks on the element. But you can always click on anything, even if it's not per say clickable. It seems to me like what we should be testing is that the clicking actually has some impact. Maybe could we test that passcode is indeed in the clipboard?

Reply:
The button click will copy the passcode to the OS clipboard.  Not something I want to check in Java code.  I also checked if I can check for some kind of event, but was not feasible without changing the involved javascript function.
This case, at least, will test the existence of the 'copybutton'.

Reply:
It looks like it's fairly easy to get the contents of the clipboard: https://stackoverflow.com/a/7106086/1296753

How would you feel about testing the contents of the clipboard just after the click, to validate that it contains the passcode?